sale_payment_sheet_financial_risk/models/res_partner.py

```diff
@@ -21,15 +23,20 @@ class ResPartner(models.Model):
         currency_field="risk_currency_id",
         compute_sudo=True,
         string="Total Sale Payment Sheet",
+        help="Sum of payment sheet amount pending to validate",
+    )
+    risk_sale_payment_sheet_info = fields.Text(
+        compute="_compute_risk_account_amount",
+        compute_sudo=True,
+        help="Sheet payment amounts grouped by reduce field (amount field in brackets)",
     )
     sale_payment_sheet_ids = fields.One2many(
         comodel_name="sale.payment.sheet.line", inverse_name="partner_id"
     )
 
     def _get_risk_sale_payment_sheet_domain(self):
-        # When p is NewId object instance bool(p.id) is False
         commercial_partners = self.filtered(
-            lambda p: (p.id and p == p.commercial_partner_id)
+            lambda p: (p.ids == p.commercial_partner_id.ids)
```

Comment:
The order is assured to be the same always? If not, do:

```suggestion
            lambda p: (set(p.ids) == set(p.commercial_partner_id.ids))
```

Reply:
p in lambda filtered is always only 1 record

Reply:
And commercial_partner_id is a Many2One field that only contains 1 record

Reply:
Then why not using `p == p.commercial_partner_id`?

Reply:
When p is NewId object instance is not equal to "Normal" registry

Modified explain message:
https://github.com/OCA/credit-control/pull/204/files#diff-9407b845adfe69cef43bd1306655f899ac7ac40987e58638425c2790365ae459R38

Reply:
OK then